#!/usr/bin/env python
PACKAGE = "pyramid_control"

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("takeoff", bool_t, 0, "takeoff", False)

gen.add("x", double_t, 0, "Desired_position (x)", 0.0, -10.0, 10.0)
gen.add("y", double_t, 0, "Desired_position (y)", 0.0, -10.0, 10.0)
gen.add("z", double_t, 0, "Desired_position (z)", 1.0, 0.0, 10.0)
gen.add("roll", double_t, 0, "Desired_attitude (roll)", 0.0, -pi, pi)
gen.add("pitch", double_t, 0, "Desired_attitude (pitch)", 0.0, -pi, pi)
gen.add("yaw", double_t, 0, "Desired_attitude (yaw)", 0.0, -pi, pi)

trajectory_modes = gen.enum([gen.const("posatt", str_t, "posatt", ""),
                             gen.const("rollpitch", str_t, "rollpitch", ""),
                             gen.const("elliptic", str_t, "elliptic", "")],
                             "Trajectory modes")

gen.add("trajectory_mode", str_t, 0, "Trajectory mode", "posatt", edit_method=trajectory_modes)

gen.add("semiMinorAxis", double_t, 0, "semiMinorAxis", 1.0, 0.0, 3.0)
gen.add("semiMajorAxis", double_t, 0, "semiMajorAxis", 1.0, 0.0, 3.0)

exit(gen.generate(PACKAGE, "pyramid_control", "TrajectoryGenerator"))
